pneumonia_detection.py

```diff
@@ -0,0 +1,141 @@
+    blob = cv2.dnn.blobFromImage(
+        image, 1 / 255.0, (416, 416), swapRB=False, crop=False
+    )
+    net.setInput(blob)
+
+    start = time.time()
+    layer_outputs = net.forward(ln)
+    end = time.time()
+
+    # Initialize our lists of detected bounding boxes, confidences
+    boxes = list()
+    confidences = list()
+    final_boxes = list()
+    final_boxes.append(image_path.split(sep="/")[-1].split(sep=".")[0] + ",")
+    for output in layer_outputs:
+        for detection in output:
+            # If pneumonia have been detect, returns the center (x, y)-coordinates
+            # of the box, followed by the width, the height and finally the confidence
+            if detection[5] > confidence:
+                # Transform coordinates from relative size to image size
+                box = detection[0:4] * np.array([w, h, w, h])
+                (center_x, center_y, width, height) = box.astype("int")
+                # Use the center (x, y)-coordinates to derive the top
+                # and left corner of the bounding box
+                x = int(center_x - (width / 2))
+                y = int(center_y - (height / 2))
+                # Update our list of bounding box coordinates, confidences,
+                # and class IDs
+                boxes.append([x, y, int(width), int(height)])
+                confidences.append(float(detection[5]))
+    # Apply non-maxima suppression to suppress weak, overlapping bounding
+    # boxes
+    idxs = cv2.dnn.NMSBoxes(boxes, confidences, confidence, threshold)
+    # Ensure at least one detection exists
+    if len(idxs) > 0:
+        # Loop over the indexes we are keeping
+        for i in idxs.flatten():
+            # Extract the bounding box coordinates
+            (x, y) = (boxes[i][0], boxes[i][1])
+            (w, h) = (boxes[i][2], boxes[i][3])
+            final_boxes.append(
+                str(round(confidences[i], 2)) + " " + str(x) + " " + str(y)
+                + " " + str(w) + " " + str(h) + " "
+            )
+            if show:
+                # Draw a bounding box rectangle and label on the image
+                color = (0, 0, 255)
+                cv2.rectangle(image, (x, y), (x + w, y + h), color, 2)
+                text = "Pneumonia: {:.4f}".format(confidences[i])
+                cv2.putText(
+                    image, text, (x, y - 5), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2
+                )
+                # Show the output image
+                cv2.imshow("Image", image)
+
+    logging.info(
+        f"Processing time: {round(end - start, 2)}seconds "
+        f"- {len(idxs)} object(s) detected"
+    )
+
+    return ''.join(final_boxes)
+
+
+def image_detection(cfg_path, images, output_path, args):
+    "Lauch detection for images"
+    result = list()
+    result.append("patientId,PredictionString")
+
+    for image in images:
+        logging.info(f"{images.index(image)+ 1} / {len(images)}")
+        box = detect(
```

Comment:
This is misleading: you don't have a box but a string representation of it !

Reply:
I changed the name with "box_coordinate"